HierarchicalDataConverter/Catalyst.HierarchicalDataConverter.AutomatedTests/GrandparentParentChildTest.cs

```diff
@@ -0,0 +1,441 @@
+                                   new Field
+                                       {
+                                           FieldName = "Level1Entity__Level1FieldToBeAdded",
+                                           Status = FieldStatus.Active
+                                       },
+                                   new Field
+                                       {
+                                           FieldName = "Level2Entity__Level2EntityPrimaryKey",
+                                           Status = FieldStatus.Active
+                                       },
+                                   new Field
+                                       {
+                                           FieldName = "Level2Entity__Level2OmittedField",
+                                           Status = FieldStatus.Omitted
+                                       },
+                                   new Field
+                                       {
+                                           FieldName = "Level2Entity__Level2FieldToBeAdded",
+                                           Status = FieldStatus.Active
+                                       }
+                               }
+                       };
+        }
+
+        private Entity GetLevel0SourceEntity()
+        {
+            return new Entity
+                       {
+                           Id = 0,
+                           EntityName = "Level0Entity",
+                           DataMartId = this.DataMartId,
+                           Fields = { new Field { IsPrimaryKey = true, FieldName = "Level0EntityPrimaryKey" } },
+                           DatabaseName = "MyDatabaseName",
+                           SchemaName = "Level0TableName"
+                       };
+        }
+
+        private Entity GetLevel1SourceEntity()
+        {
+            return new Entity
+                       {
+                           Id = 1,
+                           EntityName = "Level1Entity",
+                           DataMartId = this.DataMartId,
+                           Fields = { new Field { IsPrimaryKey = true, FieldName = "Level1EntityPrimaryKey" } },
+                           DatabaseName = "MyDatabaseName",
+                           SchemaName = "Level1TableName"
+            };
+        }
+
+        private Entity GetLevel2SourceEntity()
+        {
+            return new Entity
+                       {
+                           Id = 2,
+                           EntityName = "Level2Entity",
+                           DataMartId = this.DataMartId,
+                           Fields = { new Field { IsPrimaryKey = true, FieldName = "Level2EntityPrimaryKey" } },
+                           DatabaseName = "MyDatabaseName",
+                           SchemaName = "Level2TableName"
+            };
+        }
+
+        private int DataMartId
+        {
+            get
+            {
+                return 123;
+            }
+        }
```

Comment:
you can do "private const int DataMartId = 123;" or "private int DataMartId => 123;" if you want to be more concise

Reply:
Good idea.